### PR TITLE
`PFDisplacedVertexCandidateFinder`: fix deltaPhi cut, don't use std::abs

### LINKED ISSUE #38068: Phi wraparound bug in PFDisplacedVertexCandidateFinder?
Hi, 

I think there is a phi wraparound bug in this line of PFDisplacedVertexCandidateFinder:

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/RecoParticleFlow/PFTracking/src/PFDisplacedVertexCandidateFinder.cc#L194

```
std::abs(phi1 - phi2) > 1
```

should be

```
math::deltaPhi(phi1,phi2) > 1
```

This may be related to the issue reported [here](https://github.com/cms-sw/cmssw/issues/31158). Can someone confirm? We can fix it (and hopefully the other issue as well) if this is wrong as I expect. 

@jpata @clacaputo @laurenhay

### PR DESCRIPTION
**Pull request by mmusich**
resolves https://github.com/cms-sw/cmssw/issues/38068

#### PR description:

Implements suggestion at https://github.com/cms-sw/cmssw/issues/38068#issue-1246988087.
   * ~possibly solve https://github.com/cms-sw/cmssw/issues/31158~ (EDIT see: https://github.com/cms-sw/cmssw/pull/38080#issuecomment-1138267381)

#### PR validation:

`cmssw` compiles, waiting for a more thorough validation from PF group.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
enable profiling

**Comment by mmusich** [comment]
type pf

**Comment by jpata** [comment]
@cmsbuild please test

File: RecoParticleFlow/PFTracking/src/PFDisplacedVertexCandidateFinder.cc [hunk 1]
```diff
@@ -191,7 +192,7 @@ void PFDisplacedVertexCandidateFinder::link(const TrackBaseRef& el1,
     dist = -1;
     return;
   }
-  if (pt1 > 2 && pt2 > 2 && std::abs(phi1 - phi2) > 1) {
+  if (pt1 > 2 && pt2 > 2 && ::deltaPhi(phi1, phi2) > 1) {
```
Comment (slava77) [inline RecoParticleFlow/PFTracking/src/PFDisplacedVertexCandidateFinder.cc:195]: IIUC deltaPhi is signed. So, an abs is still needed

Comment (mmusich) [reply RecoParticleFlow/PFTracking/src/PFDisplacedVertexCandidateFinder.cc]: done in the last push.

**Comment by jpata** [comment]
baseline
https://cmssdt.cern.ch/SDT/cgi-bin/igprof-navigator/CMSSW_12_5_X_2022-05-24-2300/el8_amd64_gcc10/profiling/11834.21/igprofCPU_step3/16
Screenshot 2022-05-26 at 10 52 53

PR
https://cmssdt.cern.ch/SDT/cgi-bin/igprof-navigator/CMSSW_12_5_X_2022-05-24-2300/el8_amd64_gcc10/profiling/11834.21/PR-7c08e5/25002/igprofCPU_step3/16
Screenshot 2022-05-26 at 10 53 11


Although it's a reasonable physics fix, it looks like the timing is not appreciably different. Maybe some tuning of cuts in the PFDisplacedVertexProducer is still in order?

**Comment by mmusich** [comment]
> Although it's a reasonable physics fix, it looks like the timing is not appreciably different. Maybe some tuning of cuts in the PFDisplacedVertexProducer is still in order?

certainly it looks like it is, but I would let this to be handled by the PF group (@rappoccio mentioned in https://github.com/cms-sw/cmssw/issues/38068#issuecomment-1137258459 that a fix was available?). I was just curious to see if this relatively easy change could have mitigated the problem and it looks like it's not.
If it's preferred to have both the physics fix and the computational fix in the same PR I will gladly close this one.

**Comment by perrotta** [comment]
@mmusich @jpata I think that the fix is uncontroversial. the code currently in CMSSW being simply wrong.
I'd suggest that the computational optimization will be investigated separately

**Comment by jpata** [comment]
+reconstruction
- bugfix for https://github.com/cms-sw/cmssw/issues/38068 (addresses the physics bug)
- minor physics changes in PF and downstream
- I agree with Andrea above that this fix could go in as-is, and the investigations by the PF group into the computational performance of displaced vertices continue

**Comment by perrotta** [comment]
+1

**Comment by rappoccio** [comment]
Yes, @laurenhay and I went through this for a few hours last week but still have not discovered the other bug. Stay tuned, we will work on it more this week.

**Comment by hatakeyamak** [comment]
I doubt this is a bug per se. It's just that linking criteria is currently loose, and given large number of tracks, we are connecting too many tracks into a leading block (displaced vertex candidate). So, I think the easiest route is to find the effective criteria to tighten links.

**Comment by rappoccio** [comment]
Hi, @hatakeyamak I'm not convinced that tightening track cuts will reduce the number of tracks from 1500 to 4. Considering that this is *always* in the first event, I think there really is a bug. We'll keep looking at it.

**Comment by hatakeyamak** [comment]
From the current way of how iteratively tracks get linked, the observed certainly-undesired feature is not too surprising to me at least. Probably we can look at this from two different angles.

**Comment by rappoccio** [comment]
Could be. We can check.

**Pull request merged**